raster-foundry/raster-foundry
# Use separate DBs for each test suite

## Overview

This PR makes it so that each db test suite uses a separate database.

Implementation wise this PR:
- moves JDBC and Hikari config to a case class, which allows us to have more control over the transactor specifics -- this was necessary to allow constructor transactors that pointed to different databases
- builds up the functionality of the DBTestConfig trait so that it handles setting up the test template DB and also creating and dropping test suite specific DBs as necessary
- updates `ReadStacFeature` to use the new transactor construction (and also ran scalafmt on it)
- adds a mostly empty `.scalafmt.conf` file which makes metals happy and allows scalafmt to run within VSCode and probably other editors as well
- bumps the size of the `POSTGRES_DB_POOL` param to a default of 32 instead of 5, and also uses a pool size of 100 for tests

The DBConfig runs a one time setup routine that creates a template database on which all migrations are run. Each test-suite specific DB is created using this DB as a template so that migrations only need to be run once for each test run instead of for each test-suite.

From what I can tell, build time is not affected.

### Checklist


### Notes

I haven't seen any deadlocks yet...maybe this resolves it as a side-effect.

I expect that we'll only find out if this resolves things once it gets merged and we have a larger number of tests running.

## Testing Instructions

- CI
- Start the api and backsplash servers (after assembling) and use the app a bit to make sure everything still works (because the base transactors for each were modified), though it should have all been equivalent

Closes #3230
